x/evm/precompile/errors.go

```diff
@@ -34,14 +33,6 @@ func ErrMethodCalled(method *gethabi.Method, wrapped error) error {
 	return fmt.Errorf("%s method called: %w", method.Name, wrapped)
 }
 
-// Check required for transactions but not needed for queries
-func assertNotReadonlyTx(readOnly bool, isTx bool) error {
-	if readOnly && isTx {
-		return errors.New("cannot write state from staticcall (a read-only call)")
```

Comment:
You can add the method name as an argument to keep the error messages consistent across each tx method using the same function